packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -251,9 +252,18 @@ class MenuItem extends AutoControlledComponent<WithAsProp<MenuItemProps>, MenuIt
         </ElementType>
       </Ref>
     )
-    const maybeSubmenu =
-      menu && active && menuOpen ? (
-        <>
+    const submenuVisible = menu && active && menuOpen
+
+    // TODO: allow the users to define these via props
+    const maybeSubmenu = menu ? (
+      <>
+        <Animation
+          visible={submenuVisible}
+          timeout={2000}
```

Comment:
Should be this timeout defined in the theme?

Reply:
I believe so, we should have a way of defining timeouts in the theme in some structured way and reuse those (maybe in a similar way as we have some required icons... Haven't thought much about this, but generally I agree) But would not tackle this for now, as the `MenuItem` changes will not go as part of this PR, it is just a showcase of how it can be used and a prove of concept that it works. Good call though!